engine/cdn/item_logs_handler.go

```diff
@@ -66,27 +67,52 @@ func (s *Service) getItemLogsHandler() service.Handler {
 	}
 }
 
-func (s *Service) getItemLogsDownloadHandler() service.Handler {
+func (s *Service) getItemLogsLinesHandler() service.Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
 		vars := mux.Vars(r)
 		itemType := sdk.CDNItemType(vars["type"])
-		if err := itemType.Validate(); err != nil {
+		if err := itemType.IsLog(); err != nil {
 			return err
 		}
-		apiRef := vars["apiRef"]
-		tokenRaw := strings.TrimPrefix(r.Header.Get("Authorization"), "Bearer ")
+		token, err := s.checkItemLogsAuth(r)
+		if err != nil {
+			return err
+		}
+
+		offset, err := strconv.ParseInt(r.FormValue("offset"), 10, 64)
+		if err != nil {
+			offset = 0 // offset can be lower than 0 if we want the n last lines
+		}
+		count, err := strconv.ParseInt(r.FormValue("count"), 10, 64)
+		if err != nil || count < 0 {
+			count = 100
+		}
+
+		rc, err := s.getItemLogValue(ctx, itemType, token.APIRefHash, sdk.CDNReaderFormatJSON, offset, uint(count))
+		if err != nil {
+			return err
+		}
+		if rc == nil {
+			return sdk.WrapError(sdk.ErrNotFound, "no storage found that contains given item %s", token.APIRefHash)
+		}
 
-		// Check Authorization header
-		var token sdk.CDNAuthToken
-		v := authentication.NewVerifier(s.ParsedAPIPublicKey)
-		if err := v.VerifyJWS(tokenRaw, &token); err != nil {
-			return sdk.NewErrorWithStack(err, sdk.ErrUnauthorized)
+		return service.Write(w, rc, http.StatusOK, "application/json")
+	}
+}
+
+func (s *Service) getItemLogsDownloadHandler() service.Handler {
```

Comment:
I think we should have a more generic handler that handles CDN item download whatever their type

Reply:
I made some changes in download handler its more generic now. Filename is now human readable and generated from  values in the api ref.